modules/programs/borgmatic.nix

```diff
@@ -193,7 +218,11 @@ let
 
   writeConfig = config:
     generators.toYAML { } (removeNullValues ({
-      source_directories = config.location.sourceDirectories;
+      source_directories = if (config.location.patterns == null) then
```

Comment:
should we add an assertion to make sure that the 2 options are not used together?

Reply:
I spent some time earlier today attempting to do this, but I'm afraid I'm still rather new to Nix, so trying to define an assertion related to submodule options was giving me some trouble. I'd appreciate any pointers or suggestions you could give regarding that. In the meantime I've converted this pull request to a draft.

Reply:
Never mind, I was able to figure it out by using [autorandr](https://github.com/nix-community/home-manager/blob/master/modules/programs/autorandr.nix#L346) as an example.